src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2669,6 +2722,63 @@ _ostree_deployment_set_bootconfig_from_kargs (OstreeDeployment *deployment,
     }
 }
 
+/* Checksums overlay initrds of a deployment and updates the OstreeDeployment object. This
+ * is needed so that ostree_sysroot_write_deployments_with_options() can determine whether a
+ * bootswap is necessary when comparing deployments. */
+gboolean
+_ostree_checksum_overlay_initrds (OstreeSysroot    *self,
+                                  OstreeDeployment *deployment,
+                                  GCancellable     *cancellable,
+                                  GError          **error)
+{
+  /* On new deployments; overlay initrds are not yet written to /boot nor part of the
```

Comment:
Ownership of the overlay initrds is confusing to me.  For a libostree API user trying to add a new overlay, they own the files as long as the deployment is...staged?  Would a best practice be to use a temporary file?  (And why wouldn't we support e.g. an `O_TMPFILE`?)

What if we had a clearly distinct API like
`char *ostree_sysroot_install_overlay_initrd (OstreeSysroot *sysroot, int fd)` that would be called before creating a deployment using it?  Then ownership would be clear - create an `O_TMPFILE` with your initrd, call that API and get a checksum for it that you can pass down into the set of overlay initrds when adding another deployment.

If you call `install_overlay()` and don't call `deploy()/stage()` then they'd be GC'd, as one would expect.

Reply:
> What if we had a clearly distinct API like
`char *ostree_sysroot_install_overlay_initrd (OstreeSysroot *sysroot, int fd)` that would be called before creating a deployment using it? Then ownership would be clear - create an `O_TMPFILE` with your initrd, call that API and get a checksum for it that you can pass down into the set of overlay initrds when adding another deployment.

Hmm yeah, that sounds cleaner. Will give that a try!

Reply:
Did this now!